compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -259,7 +259,8 @@
             { ParserRuleContext.BASIC_LITERAL, ParserRuleContext.VARIABLE_REF };
 
     private static final ParserRuleContext[] TYPEDESC_RHS = {ParserRuleContext.NON_RECURSIVE_TYPE,
-            ParserRuleContext.ARRAY_TYPE_DESCRIPTOR, ParserRuleContext.OPTIONAL_TYPE_DESCRIPTOR };
+            ParserRuleContext.ARRAY_TYPE_DESCRIPTOR, ParserRuleContext.OPTIONAL_TYPE_DESCRIPTOR,
+            ParserRuleContext.UNION_TYPE_DESCRIPTOR };
```

Comment:
I feel `ParserRuleContext.UNION_TYPE_DESCRIPTOR` is not needed at all. Looks like it is not being used for any decision making within the error recovery. Only pushed and popped from the stack.

Can we use `ParserRuleContext.PIPE` here?

Reply:
Yeh seems like we can do that. But It will affect readability.

Reply:
Used ParserRuleContext.PIPE and removed ParserRuleContext.UNION_TYPE_DESCRIPTOR